ironfish/src/blockchain/blockchain.ts

```diff
@@ -969,56 +971,6 @@ export class Blockchain {
     })
   }
 
-  async addNote(index: number, note: NoteEncrypted, tx?: IDatabaseTransaction): Promise<void> {
-    return this.db.withTransaction(tx, async (tx) => {
-      const noteCount = await this.notes.size(tx)
-
-      // do we have a note at this index already?
-      if (index < noteCount) {
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const oldNote = (await this.notes.get(index, tx))!
-        if (!note.equals(oldNote)) {
-          const message = `Tried to insert a note, but a different note already there for position ${index}`
```

Comment:
Was trying to think of any cases where this assertion would happen. I'm wondering why this doesn't happen when we truncate the tree when disconnecting a block and then try to add another block. When a fork happens for instance

Reply:
Ok I see we were kind of checking the bounds when we did 
```javascript
const noteCount = await this.notes.size(tx)
```